src/main/java/org/assertj/core/api/ComparatorBasedObjectAssert.java

```diff
@@ -0,0 +1,363 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.util.Comparator;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import org.assertj.core.description.Description;
+import org.assertj.core.presentation.Representation;
+
+/**
+ * @param <ACTUAL> 
+ *
+ * @author Stefano Cordio
+ * @since 3.19.0
+ */
+public class ComparatorBasedObjectAssert<ACTUAL> extends ObjectAssert<ACTUAL>
+    implements ComparisonBasedAssert<ComparatorBasedObjectAssert<ACTUAL>, ACTUAL> {
+
+  public ComparatorBasedObjectAssert(ACTUAL actual, Comparator<? super ACTUAL> comparator, String comparatorDescription) {
+    super(actual);
+  }
+
+  @Override
+  public ComparatorBasedObjectAssert<ACTUAL> isEqualByComparingTo(ACTUAL other) {
+    // FIXME assertion
```

Comment:
My preference would be to enhance `org.assertj.core.internal.Objects` with some comparator-based methods that work only if `comparisonStrategy` is an instance of `ComparatorBasedComparisonStrategy`

Reply:
sounds good to me